tests/kernels/test_prefix_prefill.py

```diff
@@ -35,6 +35,13 @@ def test_contexted_kv_attention(
     if torch.cuda.is_available():
         torch.cuda.manual_seed(0)
     torch.set_default_device(device)
+
+    # Need this, otherwise when we capture the graph the process for GPU 1 would run on both
+    # GPU0 and GPU1 and things would hang
+    #
+    # see also similar issue: https://github.com/Dao-AILab/flash-attention/issues/523
+    torch.cuda.set_device(device)
```

Comment:
I am confused why do we need this? Can you give a more detailed example?

Reply:
There would be an error if we run the test case in environments with 2 GPU card, the test case `test_contexted_kv_attention[cuda:0-dtype0-128-64-64]` passed, but when run 
`test_contexted_kv_attention[cuda:1-dtype0-128-64-64]` (note now it uses `cuda:1`), it failed and complains:

```python
        bin = self.cache[device][key]
        if not warmup:
>           bin.c_wrapper(
                grid_0,
                grid_1,
                grid_2,
                bin.num_warps,
                bin.num_ctas,
                bin.clusterDims[0],
                bin.clusterDims[1],
                bin.clusterDims[2],
                bin.shared,
                stream,
                bin.cu_function,
                CompiledKernel.launch_enter_hook,
                CompiledKernel.launch_exit_hook,
                bin,
                *bin.assemble_tensormap_to_arg(non_constexpr_arg_values),
            )
E           ValueError: Pointer argument (at 0) cannot be accessed from Triton (cpu tensor?)

/usr/local/lib/python3.10/dist-packages/triton/runtime/jit.py:550: ValueError
```

I'm not very clear about the root causes, but I found the same issue report in flash-attention and the fix from here: https://github.com/Dao-AILab/flash-attention/issues/523#issuecomment-1707604989, and confirmed it works.